### PR TITLE
Plugins decide for themselves whether to run (osbs-client doesn't)

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-111

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: docs/build_process.md [hunk 1]
```diff
@@ -29,9 +29,6 @@ This document mentions several components that communicate with each other:
    1. The `BuildConfig`'s source git ref names the git branch the build is from
       ― The `bump_release` configuration names the branch's commit the initial
       build is from
-   1. The `stop_autorebuild_if_disabled` configuration names the config file
```
Comment (athos-ribeiro) [inline docs/build_process.md:32]: Should this be in a separate commit?

File: osbs/build/plugins_configuration.py [hunk 2]
```diff
@@ -174,68 +174,6 @@ def render(self):
...
-    def has_tag_suffixes_placeholder(self):
-        phase = 'postbuild_plugins'
-        plugin = 'tag_from_config'
-        if not self.pt.has_plugin_conf(phase, plugin):
-            logger.debug('no tag suffix placeholder')
-            return False
-
-        placeholder = '{{TAG_SUFFIXES}}'
-        plugin_conf = self.pt.get_plugin_conf('postbuild_plugins', 'tag_from_config')
-        return plugin_conf.get('args', {}).get('tag_suffixes') == placeholder
```
Comment (athos-ribeiro) [inline osbs/build/plugins_configuration.py:186]: Why is this being removed without adding an atomic reactor counterpart? Should it be in a different commit?

Comment (rcerven) [reply osbs/build/plugins_configuration.py:186]: we don't need this, that placeholder was always in json inner template, and this wasn't providing any value, IF you don't want to plugin to run, you can just simply remove it from template

File: docs/build_process.md (hunk 1 above)
Comment (rcerven) [reply docs/build_process.md:32]: not sure if it is worth, just obsolete plugin in doc

File: osbs/build/plugins_configuration.py (hunk 2 above)
Comment (athos-ribeiro) [reply osbs/build/plugins_configuration.py:186]: This seems to be unrelated to the commit that is moving code to atomic-reactor. Could it be in a separate commit? Alternatively, you could add a longer description to the commit message (e.g. bullets) explaining this specific change and the removal of `stop_autorebuild_disabled` saying both are a consequence of this change.

Comment (rcerven) [reply osbs/build/plugins_configuration.py:186]: updated commit msg

**Comment by rcerven** [comment]
rn updated: Plugins decide for themselves whether to run (osbs-client doesn't
